### PR TITLE
Port to C#

### PR DESCRIPTION
**Pull request by K2017**
Hello,

I decided to port this helper as I had to use it in a personal project involving C#.

This PR includes a new workflow for building and testing the C# version, as well as a port of the test suite to C#.

Please feel free to make any changes as you see fit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,6 +118,45 @@ def nextToken(self):
...
+        denter = DenterHelper.Builder()
+            .Nl(NL)
+            .Indent(MyCoolParser.INDENT)
+            .Dedent(MyCoolParser.DEDENT)
+            .PullToken(base.NextToken);
+    }
+
+    return denter
+        .NextToken();
+}
```
Comment (yshavit) [inline README.md:154]: minor/formatting: this brace should be indented by one

File: README.md [hunk 2]
```diff
@@ -118,6 +118,45 @@ def nextToken(self):
 NL: ('\r'? '\n' ' '*); #For tabs just switch out ' '* with '\t'*
 ```
 
+Usage (CSharp)
+=====
+
+![Build/Tests](https://github.com/yshavit/antlr-denter/workflows/CSharp/badge.svg)
+
```
Comment (yshavit) [inline README.md:125]: Please feel free to give yourself credit here, [as in the Python3 port](https://github.com/yshavit/antlr-denter/blame/master/README.md#L85). :-)

**Review by yshavit — APPROVED** [review]
Left a couple minor comments -- but feel free to merge without addressing if you want, and if so I can just do them myself afterwards. Thanks so much for the contribution!

File: README.md (hunk 1 above)
Comment (K2017) [reply README.md:156]: I found that the generated code was indented too far if indented here, not sure what to do about that

**Comment by K2017** [comment]
I think merging will be up to you, I don't have write access ;-)

File: README.md (hunk 1 above)
Comment (yshavit) [reply README.md:156]: Oh, interesting. Does that cause errors in C# (with standard linters, etc)? If so, we should write a comment so people don't get thrown off -- if not, this is fine.

**Pull request merged**